barcode-gen.py

```diff
@@ -17,6 +18,14 @@
     if len(code) != 8:
         raise Exception("Invalid code length: {}, should be 8 digits long".format(code))
 
+logo_file = os.path.join(os.path.dirname(__file__), "art", "SourceBots.svg")
+if os.path.exists(logo_file):
+    with open(logo_file, "r") as f:
+        logo_content_lines = [line for line in f if not line.strip().startswith(("<?", "<!"))]
+    logo_content = "".join(logo_content_lines)
```

Comment:
This means the code silently fails if there's no logo available. I'd argue this is a bad thing.

Reply:
That was my intention (that the logo is considered an optional feature), but I agree that silently ignoring it if it can't find the file isn't ideal.

How requiring the logo by default, but adding an option to turn it off?